hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkDataSetProcessor.java

```diff
@@ -382,25 +388,57 @@ public Partitioner getPartitioner(DataSet<ExecRow> dataSet, ExecRow template, in
     }
 
     @Override
-    public StructType getExternalFileSchema(String storedAs, String location) throws StandardException {
+    public StructType getExternalFileSchema(String storedAs, String location, boolean mergeSchema) throws StandardException {
         StructType schema = null;
+        Configuration conf = HConfiguration.unwrapDelegate();
+        FileSystem fs = null;
+        Path temp = null;
+        // normalize location string
+        location = new Path(location).toString();
         try {
+
+            if (!mergeSchema) {
+                fs = FileSystem.get(URI.create(location), conf);
+                String fileName = getFile(fs, location);
+                if (fileName != null) {
+                    temp = new Path(location, "_temp");
+                    fs.mkdirs(temp);
+                    SpliceLogUtils.info(LOG, "created temporary directory %s", temp);
+
+                    // Copy a data file to temp directory
+                    int index = fileName.indexOf(location);
+                    if (index != -1) {
+                        String s = fileName.substring(index + location.length() + 1);
+                        Path destDir = new Path(temp, s);
+                        FileUtil.copy(fs, new Path(fileName), fs, destDir, false, conf);
+                        location = temp.toString();
+                    }
+                }
+            }
             try {
+                Dataset dataset = null;
                 if (storedAs != null) {
                     if (storedAs.toLowerCase().equals("p")) {
-                        schema = SpliceSpark.getSession().read().parquet(location).schema();
+                        dataset = SpliceSpark.getSession().read().parquet(location);
                     } else if (storedAs.toLowerCase().equals("a")) {
-                        schema = SpliceSpark.getSession().read().format("com.databricks.spark.avro").load(location).schema();
+                        dataset = SpliceSpark.getSession().read().format("com.databricks.spark.avro").load(location);
                     } else if (storedAs.toLowerCase().equals("o")) {
-                        schema = SpliceSpark.getSession().read().orc(location).schema();
+                        dataset = SpliceSpark.getSession().read().orc(location);
                     }
                     if (storedAs.toLowerCase().equals("t")) {
                         // spark-2.2.0: commons-lang3-3.3.2 does not support 'XXX' timezone, specify 'ZZ' instead
                         schema = SpliceSpark.getSession().read().option("timestampFormat", "yyyy-MM-dd'T'HH:mm:ss.SSSZZ").csv(location).schema();
                     }
+                    dataset.printSchema();
```

Comment:
Is this line for debugging purpose? Do we want this to be printed out unconditionally?

Reply:
Yes, this should not be too verbose, and will be helpful trouble shooting schema mismatch problem.